perun-core/src/main/java/cz/metacentrum/perun/core/impl/MembersManagerImpl.java

```diff
@@ -689,21 +690,63 @@ public MemberGroupStatus getUnifiedMemberGroupStatus(PerunSession sess, User use
 	}
 
 	@Override
-	public Paginated<Member> getMembersPage(PerunSession sess, Vo vo, MembersPageQuery query) {
-		return jdbc.query(
-				"SELECT " + memberMappingSelectQuery +
+	public Paginated<Member> getMembersPage(PerunSession sess, Vo vo, MembersPageQuery query, String searchString, List<Status> statuses) {
+
+		String userNameQueryString = Utils.prepareUserSearchQuerySimilarMatch();
+
+		Map<String, List<String>> attributesToSearchBy = Utils.getDividedAttributes();
+
+		Map<String, Pair<String, String>> attributesToSearchByQueries = Utils.getAttributesQuery(attributesToSearchBy.get("memberAttributes"), attributesToSearchBy.get("userAttributes"), attributesToSearchBy.get("uesAttributes"));
+
+		MapSqlParameterSource namedParams = Utils.getMapSqlParameterSourceToSearchUsersOrMembers(searchString, attributesToSearchBy);
+		namedParams.addValue("voId", vo.getId());
+		namedParams.addValue("offset", query.getOffset());
+		namedParams.addValue("limit", query.getPageSize());
+
+		String statusesQueryString = "";
+		if (!statuses.isEmpty()) {
+			statusesQueryString = " members.status in (:statuses) and ";
+			List<Integer> statusCodes = statuses.stream()
+				.map(Status::getCode)
+				.collect(Collectors.toList());
+			namedParams.addValue("statuses", statusCodes);
+		}
+
+		String idQueryString = "";
+		try {
+			int id = Integer.parseInt(searchString);
+			idQueryString = " members.user_id=" + id + " or members.id=" + id + " or ";
+		} catch (NumberFormatException e) {
+			// IGNORE wrong format of ID
+		}
+
+		Paginated<Member> members = namedParameterJdbcTemplate.query(
+				"SELECT DISTINCT " + memberMappingSelectQuery +
 				query.getSortColumn().getSqlSelect() +
 				", count(*) OVER() AS total_count" +
 				" FROM members " +
+				" LEFT JOIN users ON members.user_id=users.id " +
+				" LEFT JOIN user_ext_sources ues ON ues.user_id=users.id " +
 				query.getSortColumn().getSqlJoin() +
-				" WHERE members.vo_id = ?" +
+				attributesToSearchByQueries.get("memberAttributesQuery").getLeft() +
+				attributesToSearchByQueries.get("userAttributesQuery").getLeft() +
+				attributesToSearchByQueries.get("uesAttributesQuery").getLeft() +
+				" WHERE members.vo_id = (:voId) AND " +
+				statusesQueryString +
+				" ( " +
+				" LOWER(ues.login_ext) = LOWER(:searchString) OR " +
+				attributesToSearchByQueries.get("memberAttributesQuery").getRight() +
+				attributesToSearchByQueries.get("userAttributesQuery").getRight() +
+				attributesToSearchByQueries.get("uesAttributesQuery").getRight() +
+				idQueryString +
+				userNameQueryString +
+				" ) " +
 				" ORDER BY " + query.getSortColumn().getSqlOrderBy(query) +
-				" OFFSET ?" +
-				" LIMIT ?;",
-				getPaginatedMembersExtractor(query),
-				vo.getId(),
-				query.getOffset(),
-				query.getPageSize());
+				" OFFSET (:offset)" +
+				" LIMIT (:limit);",
```

Comment:
I don't think we end queries with `;`, since its done by the DB connection logic.

Reply:
Removed.